compose/project.py

```diff
@@ -592,8 +592,17 @@ def pull_service(service):
                 service.pull(ignore_pull_failures, silent=silent)
 
     def push(self, service_names=None, ignore_push_failures=False):
+        unique_images = []
         for service in self.get_services(service_names, include_deps=False):
-            service.push(ignore_push_failures)
+            # Considering <image> and <image:latest> as the same
+            image_name_array = service.image_name.split(":")
```

Comment:
You should use [parse_repository_tag()](https://github.com/docker/compose/blob/2b604c1e8b0df4cf0da0b657634544218fc3a79d/compose/service.py#L1397-L1421)

```python
repo, tag, sep = parse_repository_tag(service.image_name)
service_image_name = sep.join(repo, tag) if tag else sep.join(repo, 'latest')
```

Reply:
Cool! I was looking for something to parse it but didn't see that.